repo/format/format_provider.go

```diff
@@ -99,11 +100,19 @@ func NewFormattingOptionsProvider(f *ContentFormat, formatBytes []byte) (Provide
 		return nil, errors.Wrap(err, "unable to create hash")
 	}
 
-	e, err := encryption.CreateEncryptor(f)
+	encryptionEncryptor, err := encryption.CreateEncryptor(f)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to create encryptor")
 	}
 
+	// This will return nil if the current options disable ECC
```

Comment:
i would add if statement around here and rename things slightly for readability:

if f.GetECCAlgorithm() != "" {
  ecc, err := ecc.New(f)
  if err ... 
  e = encryptorWrapper{e, ecc}
}

I would move `encryptorWrapper` to this package.

Reply:
Done. A side comment: i understand why you want to hide encryptorWrapper in format package, but for me it fells better to keep it together with Encryptor, because they seem very closely related.